display_list/effects/dl_image_filter.h

```diff
@@ -700,7 +700,8 @@ class DlLocalMatrixImageFilter final : public DlImageFilter {
   SkRect* map_local_bounds(const SkRect& input_bounds,
                            SkRect& output_bounds) const override {
     if (!image_filter_) {
-      return nullptr;
+      output_bounds = input_bounds;
+      return &output_bounds;
```

Comment:
I should probably commit these 3 changes separately since they aren't strictly required for the SaveLayer bounds pre-computation.

Reply:
See https://github.com/flutter/engine/pull/51337

Reply:
https://github.com/flutter/engine/pull/51340 landed with these changes and the latest commit here is rebased on top of that.